File: Atividade 01/pessoa-api-nest/src/controller/pessoa.controller.ts
import {
  Body,
  Controller,
  Delete,
  Get,
  Param,
  Post,
  Put,
} from '@nestjs/common';
import { PessoaService } from '../pessoa/pessoa.service';
import { Pessoa } from '../entitys/pessoa.entity';

@Controller('pessoas')
export class PessoaController {
  constructor(private readonly service: PessoaService) {}

  @Post()
  create(@Body() pessoa: Pessoa) {
    return this.service.salvar(pessoa);
  }

  @Get()
  findAll() {
    return this.service.listar();
  }

  @Get(':cpf')
  findOne(@Param('cpf') cpf: string) {
    return this.service.buscarPorCpf(cpf);
  }

  @Put(':cpf')
  update(@Param('cpf') cpf: string, @Body() pessoa: Pessoa) {
    return this.service.atualizar(cpf, pessoa);
  }

  @Delete(':cpf')
  remove(@Param('cpf') cpf: string) {
    return this.service.deletar(cpf);
  }
}

File: Atividade 01/pessoa-api/src/main/java/br/com/getcoders/pessoa/controller/PessoaController.java
package br.com.getcoders.pessoa.controller;

import br.com.getcoders.pessoa.infrastructure.entitys.Pessoa;
import br.com.getcoders.pessoa.service.PessoaService;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/pessoas")
@CrossOrigin(origins = "http://localhost:4200")
public class PessoaController {

    private final PessoaService service;

    public PessoaController(PessoaService service) {
        this.service = service;
    }

    @GetMapping
    public ResponseEntity<List<Pessoa>> listar() {
        return ResponseEntity.ok(service.listarTodos());
    }

    @GetMapping("/{cpf}")
    public ResponseEntity<Pessoa> buscar(@PathVariable String cpf) {
        try {
            return ResponseEntity.ok(service.buscarPorCpf(cpf));
        } catch (RuntimeException e) {
            return ResponseEntity.status(HttpStatus.NOT_FOUND).build(); // Retorna 404
        }
    }

    @PostMapping
    public ResponseEntity<Pessoa> criar(@RequestBody Pessoa pessoa) {
        return ResponseEntity.status(HttpStatus.CREATED).body(service.salvar(pessoa));
    }

    @PutMapping("/{cpf}")
    public ResponseEntity<Pessoa> atualizar(@PathVariable String cpf, @RequestBody Pessoa pessoa) {
        pessoa.setCpf(cpf);
        return ResponseEntity.ok(service.salvar(pessoa));
    }

    @DeleteMapping("/{cpf}")
    public ResponseEntity<Void> deletar(@PathVariable String cpf) {
        try {
            service.deletar(cpf);
            return ResponseEntity.noContent().build();
        } catch (RuntimeException e) {
            return ResponseEntity.status(HttpStatus.NOT_FOUND).build(); // Retorna 404
        }
    }
}